Frontend/Gmfctn/src/app/modules/+badges/components/full-list-of-achievements/full-list-of-achievements.component.ts

```diff
@@ -7,54 +9,43 @@ import { Achievement } from 'src/app/shared/models/achievement';
   templateUrl: './full-list-of-achievements.component.html',
   styleUrls: ['./full-list-of-achievements.component.scss'],
 })
-export class FullListOfAchievementsComponent implements OnInit {
-  achList: Achievement[] = [
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft turbo power ',
-      description: '',
-      xp: 5,
-      time: new Date('December 17, 2005 03:24:00'),
-    },
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft turbo power',
-      description: '',
-      xp: 5,
-      time: new Date(),
-    },
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft skylark power',
-      description: '',
-      xp: 5,
-      time: new Date('March 27, 2021 20:24:00'),
-    },
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft corporate power',
-      description: '',
-      xp: 5,
-      time: new Date('March 27, 2021 22:24:00'),
-    },
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft corporate power',
-      description: '',
-      xp: 5,
-      time: new Date('March 27, 2021 22:24:00'),
-    },
-    {
-      icon: '../../../../assets/phoenix.png',
-      name: 'Exoft corporate power',
-      description: '',
-      xp: 5,
-      time: new Date('March 27, 2021 22:24:00'),
-    },
-  ];
+export class FullListOfAchievementsComponent implements OnInit, OnDestroy {
+  achList!: Achievement[];
+  noContent = false;
+  subsription = new Subscription();
+
+  constructor(private profileService: ProfileService) {}
 
   ngOnInit(): void {
-    this.sortByDate();
+    if (!this.profileService.currentUser$.value) {
+      this.subsription.add(this.profileService.getUserInfo().subscribe(user => {
+        this.achList = user.achievements ?? [];
+
+        if (this.achList.length === 0) {
+          this.noContent = true;
+
+          return;
+        }
+
+        this.sortByDate();
```

Comment:
Duplicated code could be moved to a separate method

Reply:
Fixed